src/core/polyhedral/memory_promotion_heuristic.cc

```diff
@@ -315,6 +315,48 @@ bool isCoalesced(
   return true;
 }
 
+bool isPromotableToRegisterBelowThreads(
+    const ThreadIdxxScheduleDepthState& threadIdxxScheduleDepthState,
+    const TensorReferenceGroup& group,
+    isl::union_map schedule,
+    size_t nThreads,
+    isl::union_set activePoints) {
+  auto originalAccesses = group.originalAccesses();
+
+  // Return early if more than one element needs to be stored in registers.
+  // TODO: support arrays in registers if they are only accessed with constant
+  // subscripts, e.g. if the inner loops are fully unrolled.
+  auto sizes = group.approximationSizes();
+  auto nElements =
+      std::accumulate(sizes.begin(), sizes.end(), 1, std::multiplies<size_t>());
+  if (nElements != 1) {
+    return false;
+  }
+
+  // Since this function is only supposed to be called on groups seen _below_
+  // thread mapping, all refs in the group must all have the same thread-x
+  // depth.
+  auto depth = computeThreadIdxxScheduleDepth(
+                   threadIdxxScheduleDepthState,
+                   originalAccesses.domain().intersect(activePoints)) +
+      1;
+
+  auto scheduledAccesses =
+      originalAccesses.gist_domain(originalAccesses.domain())
+          .apply_domain(schedule);
+
+  for (auto sa : isl::UnionAsVector<isl::union_map>(scheduledAccesses)) {
```

Comment:
By the time I get to this, a few weeks after last looking at shared promotion I have forgotten what structure scheduledAccesses have, mind adding a small comment about what the structure of the ```union_map``` please?